### DISCUSSION TITLE
How can I combine a list of AdminEvent for client role mapping into one?

### DISCUSSION THREAD

**Discussion by nguyendangdtd** [keycloak/keycloak#23240] [Q&A]
### Description

We are developing the EventListenerProvider which deploys inside the Keycloak as an extension to get the data back to our services server. However, we found out that when doing the role assignment for the user there is a difference in the number of adminEvent and data carried inside.

- **Assign a list of realm roles:**
![multipleRealmRoles](https://github.com/keycloak/keycloak/assets/5369738/bce29f03-3c86-4649-844b-ffd5fe16bdf7)
From our EventListenerProvider, only one event that contains a list of assigned roles was fired by Keycloak
![realmRolesCombineInto1Event](https://github.com/keycloak/keycloak/assets/5369738/2f7f89d8-cdf4-4a3b-842d-ab386d38d5f0)

- **Assign a list of client roles:**
![multipleClientRoles](https://github.com/keycloak/keycloak/assets/5369738/a235bd9b-dd7d-4871-8137-7166473bd8c2)
From our EventListenerProvider, 2 separate events for each assigned role were fired by Keycloak
![2DifferrenceEventForClientRole](https://github.com/keycloak/keycloak/assets/5369738/33c3d289-ed9b-48a5-aab1-6edf09f45be9)

Now instead of 2 separate events (case **Assign a list of client roles**)  are triggers. Is there any way to force Keycloak to combine it into one? (similar to the case **Assign a list of realm roles**). For us, One update request is made then it will help to improve the performance and easy to manage the transaction between the keycloak and our system.

NOTE: The similar thing for case select a list of roles for unassigning.

I look forward to hearing from you.

Thanks,
Nguyen Dang

**Comment by jonkoops** [comment]
Converting this to a discussion as this is not a bug or feature request.

**Reply by nguyendangdtd** [reply]
Hello, can somebody help me please?
I do not know how bring my topic to someone who can help me.
